### PR TITLE
Add methods to support REST API

### PR DESCRIPTION
**Pull request by malakaganga**
Add methods to support reroute messages from dlc
to given specific queue in Andes class.

### CODE REVIEW

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/Andes.java [hunk 1]
```diff
@@ -791,6 +843,167 @@ public synchronized void releaseDistributedTransaction(UUID sessionId) {
...
+     * @return List<Long> of message IDs moved to the DLC from the sourceQueue.
+     * @throws AndesException if an error occurs while reading messages from the database.
+     */
+    public int rerouteAllMessagesInDeadLetterChannelForQueue(String dlcQueueName, String sourceQueue, String
+            targetQueue, int internalBatchSize, boolean restoreToOriginalQueue) throws AndesException {
+
+        List<Long> currentMessageIdList;
+        Long lastMessageId = 0L;
+        int movedMessageCount = 0;
+        currentMessageIdList = getNextNMessageIdsInDLCForQueue(sourceQueue, dlcQueueName, lastMessageId,
```
Comment (milindaperera) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/Andes.java]: We can do declaration and initialization together for currentMessageIdList

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/Andes.java [hunk 2]
```diff
@@ -791,6 +843,167 @@ public synchronized void releaseDistributedTransaction(UUID sessionId) {
...
+     * @param firstMsgId   starting message id
+     * @param count        maximum num of messages to return
+     * @return List of message metadata
+     * @throws AndesException
+     */
+    public List<AndesMessage> getNextNMessageContentInDLCForQueue(final String queueName,
+                                                                          final String dlcQueueName, long firstMsgId, int count) throws AndesException {
+        List<Long> currentMessageIdList;
+        List<AndesMessage> andesMessageList = new ArrayList<>();
+        currentMessageIdList = getNextNMessageIdsInDLCForQueue(queueName, dlcQueueName, firstMsgId, count);
```
Comment (milindaperera) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/Andes.java]: We can do declaration and initialization together for currentMessageIdList

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/Andes.java [hunk 3]
```diff
@@ -791,6 +843,167 @@ public synchronized void releaseDistributedTransaction(UUID sessionId) {
...
+     * @param restoreToOriginalQueue true if the messages need to be restored to their original
+     *                               queues instead of a single target queue.
+     * @return int Number of messages that were successfully restored.
+     * @throws AndesException if the database calls to read/delete the messages/content fails.
+     */
+    public int moveMessagesFromDLCToNewDestination(List<Long> messageIds, String sourceQueue, String targetQueue,
+                                                    boolean restoreToOriginalQueue)
+            throws AndesException {
+
+        andesChannel = createChannel(new FlowControlListener() {
```
Comment (pamod) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/Andes.java]: Could we get rid of creating a channel to move the messages ? let's stick with moving them by updating the database

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/Andes.java [hunk 4]
```diff
@@ -791,6 +843,167 @@ public synchronized void releaseDistributedTransaction(UUID sessionId) {
...
+
+            @Override
+            public void disconnect() {
+
+            }
+        });
+
+        disablePubAck = new DisablePubAckImpl();
+
+        List<AndesMessageMetadata> messagesToRemove = new ArrayList<>(messageIds.size());
```
Comment (pamod) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/Andes.java]: Could the list of message ids be null ?

**Comment by sdkottegoda** [comment]
Shall we create a jira issue and point it here?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/Andes.java [hunk 5]
```diff
@@ -587,6 +627,17 @@ public long getMessageCountInDLCForQueue(String queueName, String dlcQueueName)
...
     /**
+     * Get message IDs in dlc for a queue for a given number of messages starting from the specified id.
+     *
+     * @param sourceQueue    name of the queue
+     * @param dlcQueueName   name of the dead letter channel queue
+     * @param startMessageId starting message id
+     * @param messageLimit   maximum num of messages to return in one invocation.
+     * @return List<Long> of message IDs
+     * @throws AndesException if an error occurs while reading message IDs from database.
+     */
```
Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/Andes.java]: It seems this doc comment does not belong to any method.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/Andes.java [hunk 6]
```diff
@@ -821,6 +890,148 @@ public void notifySubscriptionFlow(UUID channelId, boolean active, DisruptorEven
...
+     * @param restoreToOriginalQueue true if the messages need to be restored to their original
+     *                               queues instead of a single target queue.
+     * @return int Number of messages that were successfully restored.
+     * @throws AndesException if the database calls to read/delete the messages/content fails.
+     */
+    public int moveMessagesFromDLCToNewDestination(List<Long> messageIds, String sourceQueue, String targetQueue,
+                                                    boolean restoreToOriginalQueue)
+            throws AndesException {
+
+        //crating an andes channel to move the messages
```
Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/Andes.java]: crating -> creating

**Comment by a5anka** [comment]
We have moved the development related to version 4.x to https://github.com/wso2/message-broker repo. The core architecture has also changed. Therefore we will have to redo this to match the new architecture.

**Pull request closed without merging**